rectools/dataset/dataset.py

```diff
@@ -245,3 +246,66 @@ def get_raw_interactions(self, include_weight: bool = True, include_datetime: bo
         pd.DataFrame
         """
         return self.interactions.to_external(self.user_id_map, self.item_id_map, include_weight, include_datetime)
+
+    def filter_on_interactions_df_row_indexes(
+        self,
+        row_indexes_to_keep: np.ndarray,
+        keep_external_ids: bool = True,
+        keep_features_for_removed_entities: bool = True,
+    ) -> "Dataset":
+        """
+        Generate filtered dataset that contains only provided `row_indexes_to_keep` from original
+        dataset interactions dataframe.
+        Resulting dataset will get new id mapping for both users and items.
+
+        Parameters
+        ----------
+        row_indexes_to_keep : np.ndarray
+            Original dataset interactions df row indexes that are to be kept
+        keep_external_ids : bool, default `True`
+            Whether to keep external ids -> 2x internal ids mapping (default).
+            Otherwise internal -> 2x internal ids mapping will be created.
+        keep_features_for_removed_entities : bool, default `True`
+            Whether to keep all features for users and items that are not hot any more.
+
+        Returns
+        -------
+        Dataset
+            Filtered dataset that has only selected interactions, new ids mapping and processed features.
+        """
+        interactions_df = self.interactions.df.loc[row_indexes_to_keep]
```

Comment:
Why did you change `iloc` to `loc`? Sure it's safe? We don't pay attention to indices usually

Reply:
Btw you wrote in docstring `Original dataset interactions df row indexes` but we don't keep original indexes, so they are row numbers actually

Reply:
`filter_on_interactions_df_row_numbers`?

Changed to iloc back

Reply:
I'd name it shorter since it's a public method. E.g. `iloc_interactions`

Reply:
That's not good since we'll get a dataset from this method (not interactions). And we will write more filters for dataset in the future (like filter on features selected)
so `filter_on` beginning is very reasonable

Reply:
ok, then let's just name it `filter` (I don't really understand `on`)

Reply:
ok)
renamed to `filter_interactions_df_rows`

Reply:
You said we're going to filter features with this method in the future - let's then call it just `filter`?

Reply:
I thought there will be different `filter_<on something>` methods. 
I'm not sure if we want to do all filters in the same method. It's probably a good idea - this way we never need to call a chain of filters (which is inefficient). So you're probably right.

It's gonna be a `filter` method. And possible arguments might be (just my imagination): `interactions_df_rows`, `user_features_subset`, `item_features_subset` and so on (if any). Right?

Reply:
yes, sounds reasonable

Reply:
Fixed